### PR TITLE
Integrate finalize envs and authentication on repos

### PR DESCRIPTION
**Pull request by geaaru**
Hereinafter the list of changes:

* add support to finalizer_envs on Luet config section
* add support to authentication section inside the repositories configurable on Luet config section
* move creation of commons dirs before luet install phase

### CODE REVIEW

File: pkg/burner/luet.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 
 func copyConfig(config, rootfsWanted string, fs vfs.FS, s *schema.SystemSpec) error {
 	//	cfg, _ := fs.RawPath(s.Luet.Config)
-	repos, err := s.Luet.Repositories.Marshal()
+	repos, err := s.Luet.Marshal()
```
Comment (mudler) [inline pkg/burner/luet.go:15]:
```suggestion
	luetCfg, err := s.Luet.Marshal()
```

**Comment by geaaru** [comment]
I'm not sure about the name of the `--skip-cleanup` option. Let me know wdyt.

**Comment by geaaru** [comment]
Can we go ahead with this?

File: pkg/burner/burner.go [hunk 2]
```diff
@@ -89,7 +89,19 @@ func prepareRootfs(s *schema.SystemSpec, fs vfs.FS, tempOverlayfs string) error
...
 		}
+
+		if s.EnsureCommonDirs {
+			info(":steaming_bowl: Ensure commons dirs.")
+			ensureDirs(tempOverlayfs)
+		}
+
 	} else if len(s.Packages.Rootfs) > 0 {
+
+		if s.EnsureCommonDirs {
```
Comment (mudler) [inline pkg/burner/burner.go:100]: Disjoin this `if` and do this operation just once

**Review by mudler — APPROVED** [review]
just a minor nitpick but looks good

File: pkg/burner/burner.go (hunk 2 above)
Comment (geaaru) [reply pkg/burner/burner.go:100]: sorry, but after two months it's a bit hard to follow the speech. Is not executed one time the method prepareRootfs?
What means `do this operation just once`?

Comment (geaaru) [reply pkg/burner/burner.go:100]: Maybe the speech is: do you want to permit to define the list of the packages to install also with image? If yes we could move  EnsureCommonsDir later and only one time.
